datajoint/expression.py

```diff
@@ -968,11 +968,14 @@ def _flatten_attribute_list(primary_key, attrs):
     """
     :param primary_key: list of attributes in primary key
     :param attrs: list of attribute names, which may include "KEY", "KEY DESC" or "KEY ASC"
-    :return: generator of attributes where "KEY" is replaces with its component attributes
+    :return: generator of attributes where "KEY" is replaced with its component attributes
     """
     for a in attrs:
         if re.match(r"^\s*KEY(\s+[aA][Ss][Cc])?\s*$", a):
-            yield from primary_key
+            if primary_key:
+                yield from primary_key
+            else:
+                continue
         elif re.match(r"^\s*KEY\s+[Dd][Ee][Ss][Cc]\s*$", a):
             yield from (q + " DESC" for q in primary_key)
```

Comment:
```suggestion
            if primary_key:
                yield from (q + " DESC" for q in primary_key)
```

Reply:
Not sure I understand. I added this to handle an edge case for `QueryExpressions` that don't have primary keys ([example](https://github.com/datajoint/datajoint-python/blob/master/tests/test_relation_u.py#L60)), but still have a valid `order_by` value. If the `order_by` contains, for example, `["KEY", "first_name"]`, then `"KEY"` needs to be skipped.

Reply:
This handles `oder_by=["KEY desc", "first_name"]`

Reply:
Ah I see. The diff was a little confusing.